pylint/checkers/variables.py

```diff
@@ -606,9 +606,25 @@ def leave_setcomp(self, _):
         # do not check for not used locals here
         self._to_consume.pop()
 
+    def _consume_in_scope(self, name, target_scope):
+        """move a name from to_consume to consumed in given scope
+        """
+        if not name:
+            return
+        for to_consume, consumed, scope in self._to_consume:
+            if scope == target_scope:
+                if name in to_consume:
+                    consumed[name] = to_consume[name]
+                    del to_consume[name]
+
     def visit_functiondef(self, node):
         """visit function: update consumption analysis variable and check locals
         """
+        for arg_annotation in node.args.annotations:
```

Comment:
I don't like a general solution as this one, since using strings does not equate an import being used. Let's instead specifically apply this to typing module. What I imagine instead:

- for each string annotation, check if there is an import of that value and if the import is from the ``typing`` module, then mark the variable as consumed.